src/vcpkg/dependencies.cpp

```diff
@@ -1250,36 +1250,41 @@ namespace vcpkg::Dependencies
                 std::vector<std::string> features;
             };
 
-            // This object contains the current version within a given column.
-            // Each "string" scheme text is treated as a separate column
-            // "relaxed" versions all share the same column
+            // This object contains the current version within a given version sheme (except for the "string" scheme,
+            // there we save an object for every version)
             struct VersionSchemeInfo
             {
                 Versions::Scheme scheme;
                 const SourceControlFileLocation* scfl = nullptr;
                 Versions::Version version;
                 // This tracks a list of constraint sources for debugging purposes
                 std::vector<std::string> origins;
+                // mapping from feature name -> dependencies of this feature
                 std::map<std::string, std::vector<FeatureSpec>> deps;
 
                 bool is_less_than(const Versions::Version& new_ver) const;
             };
 
             struct PackageNode
             {
+                // Mapping from version to the newest version in the cooresponding version sheme
                 std::map<Versions::Version, VersionSchemeInfo*, VersionTMapLess> vermap;
+                // We don't know how to compare "string" versions, so we save all versions seperatly
                 std::map<std::string, VersionSchemeInfo> exacts;
+                // Within each version sheme("version", semver, date) we only track the newest version
                 Optional<std::unique_ptr<VersionSchemeInfo>> relaxed;
                 Optional<std::unique_ptr<VersionSchemeInfo>> semver;
                 Optional<std::unique_ptr<VersionSchemeInfo>> date;
-                std::set<std::string> features;
+                std::set<std::string> requested_features;
                 bool default_features = true;
                 bool user_requested = false;
 
                 VersionSchemeInfo* get_node(const Versions::Version& ver);
+                // Created/Updates(is newer than existing) the version entry for the given sheme
                 VersionSchemeInfo& emplace_node(Versions::Scheme scheme, const Versions::Version& ver);
 
                 PackageNode() = default;
+                PackageNode(bool default_features) : default_features(default_features){};
```

Comment:
```suggestion
                explicit PackageNode(bool default_features) : default_features(default_features){};
```
This should also probably be an `enum class` `DefaultFeatures::{Yes, No}`